File: 1227-number-of-equivalent-domino-pairs/1227-number-of-equivalent-domino-pairs.java
class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        int[] count = new int[100];  
        int pairs = 0;

        for (int[] domino : dominoes) {
            int a = domino[0];
            int b = domino[1];
            int key = a < b ? a * 10 + b : b * 10 + a;

            pairs += count[key];     
            count[key]++;        
        }

        return pairs;
    }
}

File: 1227-number-of-equivalent-domino-pairs/1227-number-of-equivalent-domino-pairs.py
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        hp = defaultdict(int)
        for domino in dominoes:
            key = tuple(sorted(domino))
            hp[key] += 1
        c = 0
        for val in hp.values():
            s = val*(val-1)//2
            c += s
        return c
        # cnt = 0
        # for i in range(len(dominoes)):
        #     for j in range(i+1, len(dominoes)):
        #         if dominoes[i][0] == dominoes[j][0] and dominoes[i][1] == dominoes[j][1] or dominoes[i][0] == dominoes[j][1] and dominoes[j][0] ==  dominoes[i][1]:
        #             cnt += 1
        # return cnt
